service/controller/resource/alerting/alertmanagerconfig/resource.go

```diff
@@ -40,6 +38,21 @@ type Config struct {
 	SlackApiURL    string
 }
 
+type Resource struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+
+	baseDomain     string
+	grafanaAddress string
+	installation   string
+	mimirEnabled   bool
+	opsgenieKey    string
+	pipeline       string
+	proxy          func(reqURL *url.URL) (*url.URL, error)
+	slackApiToken  string
+	slackApiURL    string
```

Comment:
Since all those fields are taken from `Config` I would rather suggest to embed the Config struct directly within `Resource` this would save code duplication and make future changes easier.

```suggestion
	config Config
```

With this all field access must be replace from
```
r.k8sClient
```
to
```
r.config.k8sClient
```

Reply:
Sure, let me redo my 10 PRs 😅